src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1157,127 +1716,155 @@ mod tests {
     }
 
     use super::*;
+
     #[test]
-    /// Runs electric pump, checks pressure OK, shut it down, check drop of pressure after 20s
-    fn blue_loop_rat_deploy_simulation() {
-        let mut rat = RamAirTurbine::new();
-        let mut rat_controller = TestRamAirTurbineController::new();
-        let mut blue_loop = hydraulic_loop("BLUE");
-        let blue_loop_controller =
-            TestHydraulicLoopController::commanding_open_fire_shutoff_valve();
-
-        let timestep = 0.05;
-        let context = context(Duration::from_secs_f64(timestep));
-        let mut indicated_airspeed = context.indicated_airspeed();
-
-        let mut time = 0.0;
-        for x in 0..1500 {
-            rat.update_position(&context.delta());
-            if time >= 10. && time < 10. + timestep {
-                println!("ASSERT RAT STOWED");
-                assert!(blue_loop.loop_pressure <= Pressure::new::<psi>(50.0));
-                rat.deployment_commanded = false;
-                assert!(rat.position == 0.);
-            }
+    fn reservoir_gives_desired_flow() {
+        let mut reservoir = Reservoir::new(
+            "GREEN",
+            Volume::new::<gallon>(5.),
+            Volume::new::<gallon>(5.),
+        );
 
-            if time >= 20. && time < 20. + timestep {
-                println!("ASSERT RAT STOWED STILL NO PRESS");
-                assert!(blue_loop.loop_pressure <= Pressure::new::<psi>(50.0));
-                rat_controller.command_deployment();
-            }
+        assert!(Volume::new::<gallon>(1.) == reservoir.get_volume(Volume::new::<gallon>(1.)));
```

Comment:
This `reservoir.get_volume()` confused me a bit. It almost sounds like a getter to me. Maybe you could call it  something along the lines of `try_take_volume()` or `take_available_volume()`? At least that's what it looks like it does to me. Feel free to correct me. :)